make/stanc

```diff
@@ -24,24 +24,40 @@ else
 endif
 
 ifeq ($(CMDSTAN_SUBMODULES),1)
-ifeq ($(STANC3),)
-bin/stanc$(EXE) :
+# bin/stanc build rules
+ifneq (,$(STANC2))
+bin/stanc$(EXE) : O = $(O_STANC)
+bin/stanc$(EXE) : CPPFLAGS_MPI =
+bin/stanc$(EXE) : LDFLAGS_MPI =
+bin/stanc$(EXE) : LDLIBS_MPI =
+bin/stanc$(EXE) : bin/cmdstan/libstanc.a
+bin/stanc$(EXE) : bin/cmdstan/stanc.o
 	@mkdir -p $(dir $@)
-ifeq ($(OS_TAG),windows)
-	$(shell echo "curl -L https://github.com/stan-dev/stanc3/releases/download/nightly/$(OS_TAG)-stanc -o bin/stanc$(EXE)")
-else
-	curl -L https://github.com/stan-dev/stanc3/releases/download/nightly/$(OS_TAG)-stanc -o bin/stanc
-	chmod +x bin/stanc
-endif
-else
+	$(LINK.cpp) $^ $(LDLIBS) $(OUTPUT_OPTION)
+
+else ifneq ($(STANC3),)
 bin/stanc$(EXE) : $(shell find $(STANC3)/src/ -type f -name '*.ml*') $(STANC#)
 	@mkdir -p $(dir $@)
 	cd $(STANC3) && echo "--- Rebuilding stanc ---\n" && dune build @install
 	cp $(STANC3)/_build/default/src/stanc/stanc.exe $@
+
+else ifneq (,$(wildcard bin/$(OS_TAG)-stanc))
```

Comment:
Just to double check that I understand the logic here. 

If neither STANC2 or STANC3 vars are set then if make finds bin/linux-stanc, bin/windows-stanc, bin/mac-stanc it copies that file to bin/stanc? 

In what case would that be used? Or was the intention that the OS specific releases would ship with bin/linux-stanc, bin/windows-stanc, bin/mac-stanc and those would be moved on `make build`? Or is this so that `make clean` can remove bin/stanc and we have a copy of it? That does seem reasonable. 

Not saying anything is wrong, just making sure I have the reasoning behind this.

Reply:
> If neither STANC2 or STANC3 vars are set then if make finds bin/linux-stanc, bin/windows-stanc, bin/mac-stanc it copies that file to bin/stanc?

the intention is that the release tarball includes all OS-specific releases - because the tarball is platform agnostic - this means that the scripts that create the tarball need to include the tagged os-specific binaries for stanc3 - @serban-nicusor-toptal  - is this doable?

and yes, need to make sure that `make clean` removes `bin/stanc` but not `bin/mac-stanc` etc.